apps/roboshield/src/components/RobotsGenerator/RobotsGenerator.tsx

```diff
@@ -0,0 +1,215 @@
+import { Box, Paper, Stack, Step, StepButton, Stepper } from "@mui/material";
+import { FC, useEffect } from "react";
+import React from "react";
+import { useState } from "react";
+
+import Delays from "@/roboshield/components/Delays";
+import Sitemaps from "@/roboshield/components/Sitemaps";
+
+import CommonBots from "@/roboshield/components/CommonBots";
+import CommonSettings from "@/roboshield/components/CommonSettings";
+import ExistingRobots from "@/roboshield/components/ExistingRobots";
+import Finish from "@/roboshield/components/Finish";
+import {
+  useGlobalState,
+  defaultState,
+} from "@/roboshield/context/GlobalContext";
+import { generateRobots } from "@/roboshield/lib/robots";
+import RichText, { Children } from "@/roboshield/components/RichText";
+
+type Props = { [key: string]: string } & {
+  steps: {
+    title: string;
+    hint?: Children;
+    blockType:
+      | "existing-robots"
+      | "delays"
+      | "path"
+      | "block-bots"
+      | "site-maps"
+      | "finish";
+  }[];
+  actions: {
+    toolTipText: string;
+    back: string;
+    continue: string;
+    copyToClipboard: string;
+    download: string;
+    fetch: string;
+    reset: string;
+  };
+};
+
+const slugComponentsMap = {
+  "existing-robots": ExistingRobots,
+  delays: Delays,
+  path: CommonSettings,
+  "block-bots": CommonBots,
+  "site-maps": Sitemaps,
+  finish: Finish,
+};
+const RobotsGenerator: FC<Props> = React.forwardRef(
+  function RobotsGenerator(props, ref) {
+    const { steps, actions } = props;
+    const [activeStep, setActiveStep] = useState(0);
+    const { state, setState } = useGlobalState();
+    const [code, setCode] = useState(state.robots || "");
+
+    const handleNext = () => {
+      setActiveStep((prevActiveStep) => prevActiveStep + 1);
+    };
+
+    const handleBack = () => {
+      setActiveStep((prevActiveStep) => prevActiveStep - 1);
+    };
+
+    const handleReset = () => {
+      setState(defaultState);
+      setActiveStep(0);
+    };
+
+    const handleNextStep = (data: any) => {
+      const newState = { ...state, ...data };
+      setState(newState);
+      handleNext();
+    };
+
+    const handleSkipToLast = (data: any) => {
+      const newState = { ...state, ...data };
+      setState(newState);
+      setActiveStep(props.steps.length - 1);
+    };
+
+    const handleStep = (step: number) => () => {
+      setActiveStep(step);
+    };
+
+    useEffect(() => {
+      const generateRobotsFile = async () => {
+        const robots = await generateRobots(state);
+        setCode(robots);
+      };
+
+      generateRobotsFile();
+    }, [state]);
+
+    const activeStepSlug:
+      | "existing-robots"
+      | "delays"
+      | "path"
+      | "block-bots"
+      | "site-maps"
+      | "finish" = steps[activeStep]?.blockType;
+    const ActiveComponent = slugComponentsMap[activeStepSlug] ?? null;
+    const { hint, ...activeComponentProps } = steps[activeStep] ?? {};
+    return (
+      <>
```

Comment:
Why `<`>? `<>`/`Fragment` Should only be used if you have more than 1 child component and you don't want an extra `div` or something. We only have one `Section` here.

PS: It's probably best to use `div` or `Box` here instead of <`>` because you want to pass ref i.e. `<Box ref={ref}>`